runtime/compiler/compile/J9SymbolReferenceTable.cpp

```diff
@@ -1429,7 +1429,11 @@ J9::SymbolReferenceTable::findOrCreateStaticSymbol(TR::ResolvedMethodSymbol * ow
       symRef->setReallySharesSymbol();
 
    TR::KnownObjectTable::Index knownObjectIndex = TR::KnownObjectTable::UNKNOWN;
-   if (resolved
+   // check for KnownObjectTable first, in JITServer mode, this will return NULL,   
+   // so we never enter the critical section
+   TR::KnownObjectTable *knot = comp()->getOrCreateKnownObjectTable();
```

Comment:
why does this need to move higher? because of the compile-time dereference below? If so the `if (knot)` check on line 1474 should be made unconditional...